fe/fe-core/src/main/java/org/apache/doris/nereids/util/ExpressionUtils.java

```diff
@@ -102,8 +104,13 @@ public static Expression combine(ExpressionType op, List<Expression> expressions
             }
         }
 
-        Optional<Expression> result =
-                distinctExpressions.stream().reduce((left, right) -> new CompoundPredicate(op, left, right));
+        Optional<Expression> result = distinctExpressions.stream().reduce((left, right) -> {
+            if (op == ExpressionType.AND) {
+                return new And(left, right);
+            } else {
+                return new Or(left, right);
+            }
+        });
```

Comment:
Due to the previous check `Preconditions.checkArgument(op == ExpressionType.AND || op == ExpressionType.OR)`, here we can safely use `new CompoundPredicate(op, left, right))` to simplify the code.

Reply:
logical and uniformly use And
logical or uniformly use Or

Reply:
OK

Reply:
```suggestion
        Optional<Expression> result =
                distinctExpressions.stream().reduce(op == ExpressionType.AND ? And::new : Or::new);
```